### PR TITLE
add postgresql support and db migration via migrate tool and fix windows makefile

### LINKED ISSUE #2: support for postgresql

### LINKED ISSUE #3: support for windows in makefile

### PR DESCRIPTION
**Pull request by sebastienfr**
add postgresql support to golang-200 training.
add support for windows in makefile

closes #2 
closes #3 

- [x] refactoring docker-compose files using version '2'
- [x] add migration tool called *migrate* to build db scheme
- [x] update docker-compose with a volume containing the db migration files
- [x] write the posgresql implementation of the dao
- [ ] update the slides accordingdly (params, docker, makefile)
- [x] update the dao branches todos

### CODE REVIEW

**Review by SiegfriedEhret — APPROVED** [review]
Super cool !

**Review by SiegfriedEhret — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -8,8 +8,19 @@
...
+	EXTENSION=.exe
+else
+	PATHSEP=:
+	FOLDERSEP=/
+	EXTENSION=""
+endif
+
 # go env vars
-GO=$(firstword $(subst :, ,$(GOPATH)))
+GO=$(firstword $(subst $(PATHSEP), ,$(GOPATH)))
```
Comment (rlespinasse) [inline Makefile]: GOPATH n'est plus rempli par défaut, pour se protéger tu peux faire ça :

https://github.com/rlespinasse/makefile-libs/blob/master/go.mk

**Review by rlespinasse — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (sebastienfr) [reply Makefile]: C'est Windows compatible ~/go ?

Comment (rlespinasse) [reply Makefile]: c'est compatible `PowerShell`

Comment (rlespinasse) [reply Makefile]: encore mieux : `$HOME/go` on Unix systems and `%USERPROFILE%\go` on Windows

à tester : 

```Makefile
ifeq ($(GOPATH),)
ifeq ($(OS),Windows_NT)
GOPATH:=%USERPROFILE%/go
else
GOPATH:=$$HOME/go
endif
endif
GO=$(firstword $(subst $(PATHSEP), ,$(GOPATH)))
```

**Comment by sebastienfr** [comment]
Makefile fixed for windows & linux with default gopath.

**Pull request merged**